### PR TITLE
SocketWritable/Readable simplfication

### PR DESCRIPTION
**Pull request by pveentjer**
The OutboundFrame is a simplification of the SocketReadable and
SocketWritable.

A OutboundFrame has only a single method; isUrgent. The read/write methods
can all be handled in the Frame-specific ReadHandler/WriterHandler
classes. E.g. the MemberRead/Write-Handlers know how to read or
write a packet. 

The SocketReadable is completely removed. It doesn't serve a purpose since
the readhandlers work with concrete frame types. No need for polymorphism.

Also the name 'Socket' is removed since frames can be exchanged
in other ways than just sockets. E.g. one could create a shared
memory IPC version.

### CODE REVIEW

**Comment by mtopolnik** [comment]
:+1:

File: hazelcast/src/main/java/com/hazelcast/nio/tcp/spinning/SpinningSocketWriter.java [hunk 1]
```diff
@@ -180,10 +180,10 @@ private void configureBuffers(int size) {
         }
     }
 
-    private SocketWritable poll() {
+    private OutboundFrame poll() {
         for (; ; ) {
             boolean urgent = true;
-            SocketWritable packet = urgentWriteQueue.poll();
+            OutboundFrame packet = urgentWriteQueue.poll();
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/nio/tcp/spinning/SpinningSocketWriter.java]: Could be renamed to frame here.

File: hazelcast/src/main/java/com/hazelcast/nio/tcp/OldClientWriteHandler.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
 public class OldClientWriteHandler implements WriteHandler<Packet> {
 
     @Override
-    public boolean onWrite(Packet packet, ByteBuffer dst) throws Exception {
-        return packet.writeTo(dst);
+    public boolean onWrite(Packet frame, ByteBuffer dst) throws Exception {
+        return frame.writeTo(dst);
     }
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/nio/tcp/OldClientWriteHandler.java]: Could stay as `Packet packet`

File: hazelcast/src/main/java/com/hazelcast/nio/tcp/MemberWriteHandler.java [hunk 3]
```diff
@@ -30,7 +30,7 @@
 public class MemberWriteHandler implements WriteHandler<Packet> {
 
     @Override
-    public boolean onWrite(Packet packet, ByteBuffer dst) {
-        return packet.writeTo(dst);
+    public boolean onWrite(Packet frame, ByteBuffer dst) {
+        return frame.writeTo(dst);
     }
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/nio/tcp/MemberWriteHandler.java]: Could stay as `Packet packet`

Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/nio/tcp/MemberWriteHandler.java]: :+1: (with super minor comments)

File: hazelcast/src/main/java/com/hazelcast/nio/tcp/spinning/SpinningSocketWriter.java (hunk 1 above)
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/nio/tcp/spinning/SpinningSocketWriter.java]: Done.

File: hazelcast/src/main/java/com/hazelcast/nio/tcp/OldClientWriteHandler.java (hunk 2 above)
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/nio/tcp/OldClientWriteHandler.java]: Done

**Comment by jerrinot** [comment]
:+1: (After a discussion on this with @pveentjer)

**Comment by tombujok** [comment]
:+1:

**Pull request merged**